src/odemis/driver/tfsbc.py

```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
+"""
+Created on 11 May 2020
+
+@author: Philip Winkler
+
+Copyright © 2020 Philip Winkler, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+"""
+
+from odemis import model
+import logging
+import serial
+from pymodbus.client.sync import ModbusSerialClient
+import numpy
+import serial.tools.list_ports
+
+# Parameters for connection
+BAUDRATE = 230400
+TIMEOUT = 1
+BYTESIZE = 8
+PARITY = serial.PARITY_NONE
+STOPBITS = 1
+
+# Modbus level addresses
+SLAVE_UNIT = 2
+
+# Modbus registers
+BEAMDEFL_LX = 0  # lower x beam deflection element control
+BEAMDEFL_LY = 1  # lower y beam deflection element control
+BEAMDEFL_UX = 2  # upper x beam deflection element control
+BEAMDEFL_UY = 3  # upper y beam deflection element control
+
+RANGE_AMPS = [-42.2e3, 42.2e3]
+
+class BeamController(model.HwComponent):
+    """
+    Driver for the Thermofischer beam controller.
+    This class provides the .shift VA containing a tuple of two floats which describe
+    the x and y beam offset in m.
+    """
+
+    def __init__(self, name, role, amps_per_m, zero_offset_amps,
+                 rs485_serial_num, **kwargs):
+        """
+        :param port (str): port (e.g. /dev/ttyUSB0)
```

Comment:
`port` is not even used. I think the best would be to require this argument, and it can be a glob ("/dev/ttyACM*"). The serial number would be optional, and used to pick among the different ports. This would allow to not rely on the serial number if that's not useful. You might want to support also passing only the serial number and no "port", in which case it'd use the special function from `serial` (which is essentially a big glob).

BTW, please add a log line with the serial number of the port, so that if the user didn't specify it, it's easy to find out, and copy into the YAML file.

Reply:
I thought it was impossible to connect to the device without a serial number because there is no getVersion() command. This is why I deleted port from the argument list because I didn't see the point in keeping it as an argument if we always provide the serial number.

Is there a way not to rely on the serial number? As discussed, there is at least one other device (stage driver) using RS485.

Reply:
Well, there is another device using RS485 *in the first system that will use this driver*. It's best to keep it a little generic too. Especially for testing or debugging, it'll be easier if it's possible to just force the port. If you want to keep the logic simple, accept either a port name *or* a serial number.